ios/scripts/asc_assign_external_testflight_group.py

```
@@ -0,0 +1,313 @@
+
+
+def _find_build_id(token: str, app_id: str, build_number: str) -> str | None:
+    encoded_build = urllib.parse.quote(build_number, safe="")
+    status, body = _request(
+        token,
+        "GET",
+        f"/v1/builds?filter[app]={app_id}&filter[version]={encoded_build}"
+        f"&fields[builds]=version,processingState&limit=1",
+    )
+    if status != 200:
+        raise RuntimeError(f"build lookup HTTP {status} (code={_asc_error_code(body)})")
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)